lib/models/watcher.js

```diff
@@ -16,17 +18,65 @@ class Watcher extends CoreObject {
     logger.info('initialize %o', options);
 
     this.serving = _options.serving;
-    this.watcher = this.watcher || this.constructWatcher(options);
+
+    if (isExperimentEnabled('BROCCOLI_WATCHER')) {
+      this.watcher = this.watcher || this.constructBroccoliWatcher(options);
+    } else {
+      // This branch can be removed once BROCCOLI_WATCHER is enabled by default
+      this.watcher = this.watcher || this.constructWatcher(options);
+      this.watcher.on('change', this.didChange.bind(this));
```

Comment:
I now moved both sets of event listeners out of the construct methods. This is left outside of that method intentionally because of how the inheritance model with the ServerWatcher class works. This is something that we should address once we turn this feature flag on fully.